### ISSUE TITLE
VPN list is not showing names and cannot connect

### ISSUE THREAD

**Issue by herun1988** [mrodalgaard/alfred-network-workflow#4]

**Comment by mrodalgaard** [comment]
Hi @herun1988,

I need some more details on your setup to solve your problem. Is your VPN connection present in Mac Settings -> Network list? Is the `wifi` command showing results? Could you try to open the Alfred Workflows preferences, select Network, open the debug console (the little bug icon in the upper right corner) and check if an error is reported when you run the `vpn` command?

<img width="855" alt="screen shot 2016-11-06 at 19 08 30" src="https://cloud.githubusercontent.com/assets/659763/20040279/76b51e70-a454-11e6-9b1a-6fb0491117fd.png">

**Comment by herun1988** [comment]
Hi @mrodalgaard ,
Yes. I have VPN connections under "Network" list. Below is the Alfred log result.

`Starting debug for 'Network'

[2016-11-21 08:45:23][input.scriptfilter] <?xml version='1.0'?><items>
<item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> <item uid='' arg='' valid='' autocomplete=''><title></title><subtitle> ()</subtitle><icon>media/vpn.png</icon></item> </items>`

**Comment by xilopaint** [comment]
@herun1988 chances are you have IKEv2 VPN connections. The `scutil` command used by the Network workflow to list VPN connections doesn't work with them.

AppleScript can be used to show IKEv2 VPN connections which is far from a good solution for the issue.

**Comment by asyncee** [comment]
Currently scutil fully supports IKEv2 connections and connects / stops vpn services well.

**Comment by jpuris** [comment]
MacOS Mojave 10.14.6

Managed to get this working with following changes

file src/helpers.sh
line 144, remove `--\>.*`
```bash
  if [[ "$1" =~ \*[[:space:]]\((.*)\)[[:space:]].*--\>.*\"(.*)\".*\[(.*)\] ]]
```
becomes
```bash
  if [[ "$1" =~ \*[[:space:]]\((.*)\)[[:space:]].*\"(.*)\".*\[(.*)\] ]]
```

At this point the VPN list will be populated properly, however it will not connect (applescript fails with)
```
error "System Events got an error: Can’t get current configuration of service id \"<redacted>\" of network preferences." number -1728 from current configuration of service id "<redacted>" of network preferences
```

To fix this, I edited the applescript step that comes after `vpn` step in the workflow
1. Change language to `/bin/bash`
2. Replace code with following
```bash
VPN="{query}"
IS_CONNECTED=$(test -z `scutil --nc status "$VPN" | head -n 1 | grep Connected` && echo 0 || echo 1);
if [ $IS_CONNECTED = 1 ]; then
  scutil --nc stop "$VPN"
else
  scutil --nc start "$VPN"
fi
```

**Comment by jpuris** [comment]
pull request with fix #9
